library/src/jit_library/include/device_gemm_multiple_d.hpp

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <cstdlib>
+#include <vector>
+#include <memory>
+#include <sstream>
+#include <iterator>
+#include <numeric>
+#include "ck/solution_instances/gemm_add_add_fastgelu_instances.hpp"
+#include "ck/ck.hpp"
+#include "ck/utility/math.hpp"
+#include "ck_headers.hpp"
+
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+namespace device_gemm_multiple_d {
+
+
+struct Solution
+{
+    std::string template_str;
+    index_t block_size;
+    index_t grid_size;
+
+    Solution(std::string s, index_t b, index_t g) : template_str(s), block_size(b), grid_size(g)
+    {}
+
+    auto GetStr() const
+    {
+        return template_str;
+    }
+
+    auto GetBlockSize() const
+    {
+        return block_size;
+    }
+
+    auto GetGridSize() const
+    {
+        return grid_size;
+    }
+
+};
+
+std::string GetGemmSpec(const index_t m, 
+                        const index_t n, 
+                        const index_t k,
+                        const index_t m_per_block,
+                        const index_t n_per_block,
+                        const index_t k_per_block) 
+{
+    std::string spec = "";
+    if(math::integer_divide_ceil(m, m_per_block) * m_per_block - m != 0)
+        spec += "M";
+    if(math::integer_divide_ceil(n, n_per_block) * n_per_block - n != 0)
+        spec += "N";
+    if(math::integer_divide_ceil(k, k_per_block) * k_per_block - k != 0)
+        spec += "K";
+    if(spec == "")
+        return "ck::tensor_operation::device::GemmSpecialization::Default";
+
+    return "ck::tensor_operation::device::GemmSpecialization::" + spec + "Padding";
+}
+
+index_t GetGridSize(const index_t m, 
+                    const index_t n,
+                    const index_t m_per_block,
+                    const index_t n_per_block)
+{
+    return math::integer_divide_ceil(m, m_per_block) *
+            math::integer_divide_ceil(n, n_per_block);
+}
+
+const std::unordered_set<std::string>& get_xdlop_archs()
+{
+    static std::unordered_set<std::string> supported_archs{"gfx90a"};
+    return supported_archs;
+}
+
+struct Problem
+{
+    index_t M;
+    index_t N;
+    index_t K;
+    index_t NumDTensors;
+    bool TransA;
+    bool TransB;
+    bool TransCDE;
+    std::string ADataType;
+    std::string BDataType;
+    std::string CDEDataType;
```

Comment:
This should be named `CEDataType`. You are also missing `std::vector<std:string> DDataTypes`.